### PR TITLE
Prevent overflow in random_resized_crop tests

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

## Category:

**Other** (Bug in test code)

## Description:

- It fixes an overflow problem in test code, causing the test to fail

### Affected modules and functionalities:

test_random_resized_crop.py

### Key points relevant for the review:

NA
### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/test/python/operator/test_random_resized_crop.py [hunk 1]
```diff
@@ -20,7 +20,9 @@
 
 
 def close(a, b):
-    return abs(a - b) < 1e-5 or abs(a - b) < abs(a) + abs(b) * 1e-6
```
Comment (stiepan) [inline dali/test/python/operator/test_random_resized_crop.py:23]: I can't wrap my head around the `or` part. It is always true if a >= b >= 0.

File: dali/test/python/operator/test_random_resized_crop.py [hunk 2]
```diff
@@ -145,7 +147,8 @@ def _test_rrc(device, max_frames, layout, aspect_ratio_range, area_range, output
         input = input.gpu()
     out = fn.random_resized_crop(input, random_aspect_ratio=aspect_ratio_range,
                                  random_area=area_range, size=output_size,
-                                 interp_type=dali.types.INTERP_LINEAR, seed=12321,
+                                 interp_type=dali.types.INTERP_LINEAR,
+                                 antialias=False, seed=12321,
```
Comment (jantonguirao) [inline dali/test/python/operator/test_random_resized_crop.py:151]: this was not necessary but it silents a warning and the default behavior for INTERP_LINEAR is now different, so this restores the no-antialiasing behavior as when this testcase was introduced

File: dali/test/python/operator/test_random_resized_crop.py (hunk 1 above)
Comment (jantonguirao) [reply dali/test/python/operator/test_random_resized_crop.py:23]: actually I don't really understand the need for the second expression :/

Comment (jantonguirao) [reply dali/test/python/operator/test_random_resized_crop.py:23]: a=10, b=11
abs(a-b) -> 1
abs(a) + abs(b) * 1e-6 -> 10 + 11*1e-6 -> 10.000011
1 < 1e-5 -> false
10 < 10.000011 -> true

Comment (jantonguirao) [reply dali/test/python/operator/test_random_resized_crop.py:23]: a=10, b=30
abs(a-b) = 20
abs(a) + abs(b) * 1e-6 -> 10.00003
10 < 1e-5 -> false
20 < 10.00003 -> false

**Comment by jantonguirao** [comment]
!build

**Review by stiepan — APPROVED** [review]

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

File: dali/test/python/operator/test_random_resized_crop.py [hunk 3]
```diff
@@ -20,7 +20,8 @@
 
 
 def close(a, b):
-    return abs(a - b) < 1e-5 or abs(a - b) < abs(a) + abs(b) * 1e-6
+    absdiff = a - b if b < a else b - a
+    return absdiff < 1e-5 or absdiff < abs(a) + abs(b) * 1e-6
```
Comment (stiepan) [inline dali/test/python/operator/test_random_resized_crop.py:24]:
```suggestion
    if isinstance(a, np.float32):
        return np.isclose(a, b)
    absdiff = a - b if b < a else b - a
    return absdiff <= 1
```

Comment (jantonguirao) [reply dali/test/python/operator/test_random_resized_crop.py]: Done

**Comment by jantonguirao** [comment]
!build

File: dali/test/python/operator/test_random_resized_crop.py (hunk 1 above)
Comment (staniewzki) [reply dali/test/python/operator/test_random_resized_crop.py:23]: Yes, it should have been `(abs(a) + abs(b)) * 1e-6`

**Review by banasraf — APPROVED** [review]

**Pull request merged**